doc/versioning.rst

```diff
@@ -82,3 +82,26 @@ Example
   | oneAPI provisional 1.1 rev 2
   | oneAPI provisional 1.1 rev 3
   | oneAPI 1.1 rev 1
+
+Programmatic versioning
+=======================
+
+The specification version as mentioned above can be used programatically by appending all digits of the specification version in the format of <MAJOR><MINOR><REVISION>. MINOR and REVISION version use two digits. This versioning can be used to check the compatibility of the implementation with the specification version.
+
+Example
+  | oneAPI 1.1 rev 1 will be represented as a numerical value of 10101
+  | oneAPI 1.2 rev 1 will be represented as a numerical value of 10201
+  | oneAPI 1.2 rev 2 will be represented as a numerical value of 10202
+
+The implementation is required to define a macro with the numerical value of the specification version. The macro name should be `ONEAPI_SPEC_VERSION` and the value should be the numerical value of the specification version. The components within the specification, can implement their own macro with the numerical value of the specification version. The macro name should be `<COMPONENT>_SPEC_VERSION` and the value should be the numerical value of the specification version.
```

Comment:
> The PR title suggests that this change only affects oneMKL but it seems to be the scope of this change is broader that than.

I updated the title. You are right, it affects the full document and all components.

> What is the benefit of having ONEAPI_SPEC_VERSION  over only defining \<COMPONENT\>_SPEC_VERSION?

The spec currently does not support versioning on a component level, only oneAPI level, so the `<COMPONENT>_SPEC_VERSION` is optional, but `ONEAPI_SPEC_VERSION` is mandatory.

Reply:
A mandatory macro with the same name in different projects is a terrible idea. On the user side, it can always happen that libraries implementing a different version of the specification are used, in which case one gets macro redefinition error.

Reply:
Yes I also have concerns about defining the same macro in multiple projects.

Reply:
Updated